internal/storage/storage_handle.go

```diff
@@ -39,22 +46,57 @@ type storageClient struct {
 	client *storage.Client
 }
 
-// NewStorageHandle returns the handle of Go storage client containing
-// customized http client. We can configure the http client using the
-// storageClientConfig parameter.
-func NewStorageHandle(ctx context.Context, clientConfig storageutil.StorageClientConfig) (sh StorageHandle, err error) {
+// Followed https://pkg.go.dev/cloud.google.com/go/storage#hdr-Experimental_gRPC_API to create the gRPC client.
+func createGRPCClientHandle(ctx context.Context, clientConfig *storageutil.StorageClientConfig) (sc *storage.Client, err error) {
+	if clientConfig.ClientProtocol != mountpkg.GRPC {
```

Comment:
it's fine to remove this check as this func's responsibility is to create GRPC client. if this check has to be added, it should be added in the function calling this function.

we can do the same for http client func as well

Reply:
Added it since, it was hard to write unit test for calling method - NewStorageHandle().

Reply:
Discussed offline to keep the check in both createGRPCClientHandle & createHTTPClientHandle - to make sure, this is being called with right client-protocol.